File: 739-DailyTemperatures/739-DailyTemperatures.js
// Last updated: 24/07/2026, 23:50:35
/**
 * @param {number[]} temperatures
 * @return {number[]}
 */
var dailyTemperatures = function (tempreatures) {
    let stack = []
    let res = Array(tempreatures.length).fill(0);

    for (let i = tempreatures.length - 1; i >= 0; i--) {
        while (stack.length && tempreatures[i] >= tempreatures[stack[stack.length - 1]]) {
            stack.pop();
        }

        if (stack.length) {
            res[i] = stack[stack.length - 1] - i
        }

        stack.push(i)
        // console.log(stack)
    }

    return res
};

File: 739-DailyTemperatures/739-DailyTemperatures.java
// Last updated: 24/07/2026, 23:50:41
import java.util.*;
class Solution {
    public int[] dailyTemperatures(int[] temp) {
        int answer [] = new int[temp.length];
        Stack <Integer> stk = new Stack<Integer>();

        for(int i=temp.length-1;i>=0;i--){
            while(!stk.isEmpty()&&temp[i]>=temp[stk.peek()]){
                stk.pop();
            }

            if(!stk.isEmpty()){
                answer[i]=stk.peek()-i;
            }
            stk.push(i);
            }

        return answer;
        }
        
}